xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1650,28 +1650,28 @@ bool CGUIMediaWindow::OnPopupMenu(int itemIdx)
     int i = 0;
     while (item->HasProperty(StringUtils::Format("contextmenulabel(%i)", i)))
     {
-      buttons.emplace_back(-buttons.size(), item->GetProperty(StringUtils::Format("contextmenulabel(%i)", i)).asString());
+      buttons.emplace_back(buttons.size(), item->GetProperty(StringUtils::Format("contextmenulabel(%i)", i)).asString());
       ++i;
     }
   }
-  auto pluginMenuRange = std::make_pair(0, buttons.size());
+  auto pluginMenuRange = std::make_pair(static_cast<size_t>(0), buttons.size());
 
   //Add the global menu
   auto globalMenu = CContextMenuManager::GetInstance().GetItems(*item, CContextMenuManager::MAIN);
   auto globalMenuRange = std::make_pair(buttons.size(), buttons.size() + globalMenu.size());
   for (const auto& menu : globalMenu)
-    buttons.emplace_back(-buttons.size(), menu->GetLabel(*item));
+    buttons.emplace_back(buttons.size(), menu->GetLabel(*item));
 
   //Add legacy items from windows
-  auto windowMenuRange = std::make_pair(buttons.size(), -1);
+  auto buttonsSize = buttons.size();
   GetContextButtons(itemIdx, buttons);
-  windowMenuRange.second = buttons.size();
+  auto windowMenuRange = std::make_pair(buttonsSize, buttons.size());
 
   //Add addon menus
   auto addonMenu = CContextMenuManager::GetInstance().GetAddonItems(*item, CContextMenuManager::MAIN);
   auto addonMenuRange = std::make_pair(buttons.size(), buttons.size() + addonMenu.size());
   for (const auto& menu : addonMenu)
-    buttons.emplace_back(-buttons.size(), menu->GetLabel(*item));
+    buttons.emplace_back(buttons.size(), menu->GetLabel(*item));
```

Comment:
The `-` was probably used to prevent conflicts with other positive-ID buttons added. Probably keep the `-` and cast to int.

Reply:
yeah it probably was but it's never been working that way, removing the `-` retains the current behaviour and removes the warning so it felt like the safest fix for now.

Reply:
Are you sure about that? If you look at CContextButtons::Add there is a dupe check, so if you reuse any numbers in that range, it wont add it.

Reply:
buttons.size() returns a size_t which is unsigned so the negation does nothing and even if it did work buttons is `std::vector<std::pair<unsigned int, std::string>>` so it would never be negative.

Doing `buttons.emplace_back(-static_cast<int>(buttons.size()), menu->GetLabel(*item));` would work but result in very large index values as they would be cast to unsigned.

Reply:
I'll do a test to be sure. I found it because VS Warned about unary minus on unsigned operand is still positive

Reply:
I was wrong it seems. cpp.sh/5qzxz I'll go ahead and fix it

Reply:
Okay, replaced the negation with the bitwise negation operator instead to provide similar behavior.